src/components/duelling-picklist/duelling-picklist.test.js

```diff
@@ -0,0 +1,990 @@
+  const [notSelectedItems, setNotSelectedItems] = React.useState(allItems);
+  const [notSelectedSearch, setNotSelectedSearch] = React.useState({});
+  const [selectedItems, setSelectedItems] = React.useState({});
+  const [searchQuery] = React.useState("");
+  const isSearchMode = Boolean(searchQuery.length);
+  const onAdd = React.useCallback(
+    (item) => {
+      const { [item.key]: removed, ...rest } = notSelectedItems;
+      setNotSelectedItems(rest);
+      setSelectedItems({ ...selectedItems, [item.key]: item });
+      const { [item.key]: removed2, ...rest2 } = notSelectedSearch;
+      setNotSelectedSearch(rest2);
+    },
+    [notSelectedItems, notSelectedSearch, selectedItems]
+  );
+  const onRemove = React.useCallback(
+    (item) => {
+      const { [item.key]: removed, ...rest } = selectedItems;
+      setSelectedItems(rest);
+      setNotSelectedItems({ ...notSelectedItems, [item.key]: item });
+      if (isSearchMode && item.title.includes(searchQuery)) {
+        setNotSelectedSearch({ ...notSelectedSearch, [item.key]: item });
+      }
+    },
+    [
+      isSearchMode,
+      notSelectedItems,
+      notSelectedSearch,
+      searchQuery,
+      selectedItems,
+    ]
+  );
+  const renderItems = (list, type, handler) =>
+    order.reduce((items, key) => {
+      const item = list[key];
+      if (item) {
+        items.push(
+          <PicklistItem
+            key={key}
+            type={type}
+            item={item}
+            onChange={handler}
+            {...props}
+          >
+            <div style={{ display: "flex", width: "100%" }}>
+              <div style={{ width: "50%" }}>
+                <p style={{ fontWeight: 700, margin: 0, marginLeft: 24 }}>
+                  {item.title}
+                </p>
+              </div>
+              <div style={{ width: "50%" }}>
+                <p style={{ margin: 0 }}>{item.description}</p>
+              </div>
+            </div>
+          </PicklistItem>
+        );
+      }
+      return items;
+    }, []);
+  return (
+    <div>
+      <DuellingPicklist
+        leftLabel={`List 1 (${Object.keys(notSelectedItems).length})`}
+        rightLabel={`List 2 (${Object.keys(selectedItems).length})`}
+        disabled={isEachItemSelected}
+      >
+        <Picklist
+          disabled={isEachItemSelected}
+          placeholder={<PicklistPlaceholder text="Unassigned list empty" />}
+          {...props}
+        >
+          {renderItems(
+            isSearchMode ? notSelectedSearch : notSelectedItems,
+            "add",
+            onAdd
+          )}
+        </Picklist>
+        <Picklist
+          disabled={isEachItemSelected}
+          placeholder={<PicklistPlaceholder text="Nothing to see here" />}
+          {...props}
+        >
+          {renderItems(selectedItems, "remove", onRemove)}
+        </Picklist>
+      </DuellingPicklist>
+    </div>
+  );
+};
+
+context("Testing Duelling-Picklist component", () => {
```

Comment:
we should use: 
`Duelling-Picklist` or `Duelling Picklist` - your choice :) 

```suggestion
context("Testing Duelling-Picklist component", () => {
```